subprojects/core/src/testFixtures/groovy/org/gradle/api/tasks/AbstractSpockTaskTest.groovy

```diff
@@ -42,28 +39,28 @@ import java.util.concurrent.atomic.AtomicBoolean
 
 import static org.junit.Assert.assertFalse
 
-public abstract class AbstractSpockTaskTest extends AbstractProjectBuilderSpec {
+abstract class AbstractSpockTaskTest extends AbstractProjectBuilderSpec {
     public static final String TEST_TASK_NAME = "taskname"
 
     def taskClassInfoStore = new DefaultTaskClassInfoStore(new DefaultTaskClassValidatorExtractor())
     private final ITaskFactory taskFactory = new AnnotationProcessingTaskFactory(taskClassInfoStore, new TaskFactory(new AsmBackedClassGenerator()))
 
-    public abstract AbstractTask getTask();
+    abstract AbstractTask getTask()
 
-    public <T extends AbstractTask> T createTask(Class<T> type) {
-        return createTask(type, project, TEST_TASK_NAME);
+    def <T extends AbstractTask> T createTask(Class<T> type) {
```

Comment:
Is `def` needed here?

Reply:
Yes, it is. IDEA gives me a warning for the unused `public`, but when using `def` at least the warning is gone.